src/Functions/UserDefined/UserDefinedSQLObjectsLoaderFromZooKeeper.cpp

```diff
@@ -0,0 +1,416 @@
+    LOG_DEBUG(log, "Storing user-defined object {} at zk path {}", backQuote(object_name), path);
+
+    WriteBufferFromOwnString create_statement_buf;
+    formatAST(create_object_query, create_statement_buf, false);
+    writeChar('\n', create_statement_buf);
+    String create_statement = create_statement_buf.str();
+
+    auto zookeeper = getZooKeeper();
+
+    size_t num_attempts = 10;
+    while (true)
+    {
+        auto code = zookeeper->tryCreate(path, create_statement, zkutil::CreateMode::Persistent);
+        if ((code != Coordination::Error::ZOK) && (code != Coordination::Error::ZNODEEXISTS))
+            throw zkutil::KeeperException(code, path);
+
+        if (code == Coordination::Error::ZNODEEXISTS)
+        {
+            if (throw_if_exists)
+                throw Exception(ErrorCodes::FUNCTION_ALREADY_EXISTS, "User-defined function '{}' already exists", object_name);
+            else if (!replace_if_exists)
+                return false;
+
+            code = zookeeper->trySet(path, create_statement);
+            if ((code != Coordination::Error::ZOK) && (code != Coordination::Error::ZNONODE))
+                throw zkutil::KeeperException(code, path);
+        }
+
+        if (code == Coordination::Error::ZOK)
+            break;
+
+        if (!--num_attempts)
+            throw zkutil::KeeperException(code, path);
+    }
+    LOG_DEBUG(log, "Object {} stored", backQuote(object_name));
+
+    /// Refresh object and set watch for it. Because it can be replaced by another node after creation.
+    refreshObject(zookeeper, object_type, object_name);
+
+    return true;
+}
+
+
+bool UserDefinedSQLObjectsLoaderFromZooKeeper::removeObject(
+    UserDefinedSQLObjectType object_type, const String & object_name, bool throw_if_not_exists)
+{
+    String path = getNodePath(zookeeper_path, object_type, object_name);
+    LOG_DEBUG(log, "Removing user-defined object {} at zk path {}", backQuote(object_name), path);
+
+    auto zookeeper = getZooKeeper();
+
+    auto code = zookeeper->tryRemove(path);
+    if ((code != Coordination::Error::ZOK) && (code != Coordination::Error::ZNONODE))
+        throw zkutil::KeeperException(code, path);
+
+    if (code == Coordination::Error::ZNONODE)
+    {
+        if (throw_if_not_exists)
+            throw Exception(ErrorCodes::UNKNOWN_FUNCTION, "User-defined object '{}' doesn't exist", object_name);
+        else
+            return false;
+    }
+
+    LOG_DEBUG(log, "Object {} removed", backQuote(object_name));
+    return true;
+}
+
+bool UserDefinedSQLObjectsLoaderFromZooKeeper::getObjectDataAndSetWatch(
+    const zkutil::ZooKeeperPtr & zookeeper,
+    String & data,
+    const String & path,
+    UserDefinedSQLObjectType object_type,
+    const String & object_name)
+{
+    const auto object_watcher = [watch_queue = watch_queue, object_type, object_name](const Coordination::WatchResponse & response)
+    {
+        if (response.type == Coordination::Event::CHANGED) [[maybe_unused]]
```

Comment:
`maybe_unused` is specified at the wrong line

Reply:
Corrected this